File: leetcode/lc-us/cpp/1470.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> shuffle(vector<int> &nums, int n) {
        for (int i = n; i < 2 * n; i++) {
            nums[i] |= (nums[i - n] << 10);
        }
        for (int i = 0; i < n; i++) {
            int j = i * 2;
            nums[j] = nums[i + n] >> 10;
            nums[j + 1] = nums[i + n] & 1023;
        }
        return nums;
    }
};

File: leetcode/lc-us/py/1470.py
from typing import List


class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        res = [0] * (n * 2)
        u, i, j = 0, 0, n
        while u < len(res):
            res[u] = nums[i]
            res[u + 1] = nums[j]
            u += 2
            i += 1
            j += 1
        return res
